pkg/authn/helper_test.go

```diff
@@ -136,3 +149,61 @@ func TestBadOutput(t *testing.T) {
 		t.Errorf("Authorization() = %v", got)
 	}
 }
+
+// TestHTTPSURL checks that helper saving https works
+func TestHTTPSURL(t *testing.T) {
+	output := `{"ServerURL":"","Username":"mainuser","Secret":"FVLGCQr_a-FZJM2ON227-YXdAEJn3oZZnnWs6Jd6iVE"}`
+	f := func(cmd *exec.Cmd) error {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(cmd.Stdin)
+		s := buf.String()
+
+		var err error
+
+		if strings.HasPrefix(s, "https://foo.dev") {
+			_, err = cmd.Stdout.Write([]byte(output))
+		} else {
+			_, err = cmd.Stdout.Write([]byte(magicNotFoundMessage))
+		}
+		return err
+	}
+
+	h := &helper{name: "test", domain: testDomain, r: &customRunner{f: f}}
+	got, err := h.Authorization()
+	if err != nil {
+		t.Errorf("Authorization() = %v", got)
+	}
+
+	if got != `Basic bWFpbnVzZXI6RlZMR0NRcl9hLUZaSk0yT04yMjctWVhkQUVKbjNvWlpubldzNkpkNmlWRQ==` {
+		t.Fatalf("Authorization() returned unexepcted result = %v (expected Basic bWFpbnVzZXI6RlZMR0NRcl9hLUZaSk0yT04yMjctWVhkQUVKbjNvWlpubldzNkpkNmlWRQ==)", got)
+	}
+}
+
+// TestProtocollessURL checks that helper saving without protocol works
+func TestProtocollessURL(t *testing.T) {
+	output := `{"ServerURL":"","Username":"mainuser","Secret":"FVLGCQr_a-FZJM2ON227-YXdAEJn3oZZnnWs6Jd6iVE"}`
+	f := func(cmd *exec.Cmd) error {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(cmd.Stdin)
+		s := buf.String()
+
+		var err error
+
+		if strings.HasPrefix(s, "foo.dev") {
+			_, err = cmd.Stdout.Write([]byte(output))
+		} else {
+			_, err = cmd.Stdout.Write([]byte(magicNotFoundMessage))
+		}
+		return err
+	}
+
+	h := &helper{name: "test", domain: testDomain, r: &customRunner{f: f}}
+	got, err := h.Authorization()
+	if err != nil {
+		t.Fatalf("Authorization() = %v", got)
+	}
+
+	if got != `Basic bWFpbnVzZXI6RlZMR0NRcl9hLUZaSk0yT04yMjctWVhkQUVKbjNvWlpubldzNkpkNmlWRQ==` {
```

Comment:
Can you make it more obvious where this auth token comes from? Even just having it generated from base64-encoding consts used elsewhere will make it clearer, with a bonus that it'll be easy to change the consts in the future if we ever need to.

Reply:
Done.